src/app/child-dev-project/attendance/activity-setup/activity-setup.component.ts

```diff
@@ -0,0 +1,136 @@
+})
+export class ActivitySetupComponent implements OnInit {
+  date = new Date();
+
+  existingEvents: Note[] = [];
+
+  selectedEvent: Note;
+  @Output() eventSelected = new EventEmitter<Note>();
+
+  allActivities: RecurringActivity[] = [];
+  visibleActivities: RecurringActivity[] = [];
+
+  constructor(
+    private entityMapper: EntityMapperService,
+    private attendanceService: AttendanceService,
+    private sessionService: SessionService,
+    private formDialog: FormDialogService
+  ) {}
+
+  async ngOnInit() {
+    await this.initAvailableEvents();
+  }
+
+  private async initAvailableEvents() {
+    this.existingEvents = await this.attendanceService.getEventsOnDate(
+      this.date
+    );
+    await this.loadActivities();
+    this.sortEvents();
+  }
+
+  private async loadActivities() {
+    this.allActivities = await this.entityMapper.loadType<RecurringActivity>(
+      RecurringActivity
+    );
+    this.visibleActivities = this.allActivities.filter(
+      (a) =>
+        a.assignedTo === this.sessionService.getCurrentUser().getId() ||
+        a.assignedTo === ""
+    );
+
+    for (const activity of this.visibleActivities) {
+      this.existingEvents.push(this.createEventForActivity(activity));
+    }
+  }
+
+  showMore() {
+    const additionalActivities = this.allActivities.filter(
+      (a) => !this.visibleActivities.includes(a)
+    );
+    for (const activity of additionalActivities) {
+      this.existingEvents.push(this.createEventForActivity(activity));
+      this.visibleActivities.push(activity);
+    }
+    this.sortEvents();
+  }
+
+  async setNewDate(date: Date) {
+    this.date = date;
+
+    await this.initAvailableEvents();
+
+    if (!RecurringActivity.isActivityEventNote(this.selectedEvent)) {
+      this.selectedEvent = null;
+    }
+  }
+
+  private createEventForActivity(activity: RecurringActivity): Note {
+    const alreadyCreated = this.existingEvents.find(
+      (e) => e.relatesTo === activity._id
+    );
+    if (alreadyCreated) {
+      return alreadyCreated;
+    }
+
+    const event = Note.create(this.date, activity.title);
+    event.children = activity.participants;
+    event.relatesTo = activity._id;
+    return event;
+  }
+
+  private sortEvents() {
+    const calculateEventPriority = (event: Note) => {
+      if (!RecurringActivity.isActivityEventNote(event)) {
+        return 0;
+      }
+
+      let score = 1;
+      const activity = this.allActivities.find(
+        (a) => a._id === event.relatesTo
+      );
+      if (
+        activity.assignedTo === this.sessionService.getCurrentUser().getId()
+      ) {
+        score += 1;
+      } else if (activity.assignedTo !== "") {
+        score -= 2;
+      }
+
+      return score;
+    };
+
+    this.existingEvents = this.existingEvents.sort(
+      (a, b) => calculateEventPriority(b) - calculateEventPriority(a)
+    );
+  }
+
+  createOneTimeEvent() {
+    const newNote = new Note(Date.now().toString());
```

Comment:
Is there a rule to create notes like this? With the date as the id?

Reply:
`Note.create(new Date());` seems more elegant. will change it